test/stage1/behavior/error.zig

```diff
@@ -375,3 +375,25 @@ test "implicit cast to optional to error union to return result loc" {
     S.entry();
     //comptime S.entry(); TODO
 }
+
+const Bar = error {
+    A,
+    B,
+};
+
+test "implicit cast from enum literal to error union" {
+    var err1: Bar!u32 = .B;
+    var err2: anyerror!u32 = .RandomError;
```

Comment:
This is supposed to be a compile error. There's an ambiguity if it was `anyerror!Enum` and `enum { RandomError }`.

Reply:
This should now be solved